lib/puppet/util/package.rb

```diff
@@ -1,8 +1,13 @@
 module Puppet::Util::Package
   def versioncmp(version_a, version_b)
+    
+    # remove unnecesary ending .0 or .00.0 from version
+    trimmed_version_a = version_a.sub(/(-0+|\.0+)+$/, '')
+    trimmed_version_b = version_b.sub(/(-0+|\.0+)+$/, '')
```

Comment:
Maybe add `# frozen_string_literal: true` to avoid creating temp strings each time?